### ISSUE TITLE
rancher/rke-tools:v0.1.40-arm64 isn't built correctly

### ISSUE THREAD

**Issue by westonmyers** [rancher/rke#1553]
**RKE version:**
AMD64
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:32:14Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
ARM64
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:32:14Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/arm64"}
```
Backend
```
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:32:14Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
``` 

**Docker version:**
ARM64
```
Client:
 Version:           19.03.1-ce
 API version:       1.40
 Go version:        go1.12.7
 Git commit:        74b1e89e8ac68948be88fe0aa1e2767ae28659fe
 Built:             Fri Jul 26 22:05:37 2019
 OS/Arch:           linux/arm64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.1-ce
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.7
  Git commit:       74b1e89e8ac68948be88fe0aa1e2767ae28659fe
  Built:            Fri Jul 26 22:04:25 2019
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          v1.2.7
  GitCommit:        85f6aa58b8a3170aec9824568f7a31832878b603
 runc:
  Version:          1.0.0-rc8
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
 docker-init:
  Version:          0.18.0
  GitCommit: 
```
AMD64
```
Client:
 Version:           19.03.1-ce
 API version:       1.40
 Go version:        go1.12.7
 Git commit:        74b1e89e8ac68948be88fe0aa1e2767ae28659fe
 Built:             Fri Jul 26 22:04:55 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.1-ce
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.7
  Git commit:       74b1e89e8ac68948be88fe0aa1e2767ae28659fe
  Built:            Fri Jul 26 22:04:16 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.7
  GitCommit:        85f6aa58b8a3170aec9824568f7a31832878b603
 runc:
  Version:          1.0.0-rc8
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f
 docker-init:
  Version:          0.18.0
  GitCommit: 
```

**Operating system and kernel:**
Alpine Linux
ARM64
`4.19.65-0-virt`
AMD64
`4.19.66-0-rpi`

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AMD64
`KVM`
ARM64
`Bare-metal`

**cluster.yml file:**
```
# 
# Cluster Config
# 
answers: {}
default_pod_security_policy_template_id: unrestricted
docker_root_dir: /var/lib/docker
enable_cluster_alerting: true
enable_cluster_monitoring: true
enable_network_policy: false
local_cluster_auth_endpoint:
  enabled: true
name: myers
# 
# Rancher Config
# 
rancher_kubernetes_engine_config:
  addon_job_timeout: 30
  authentication:
    strategy: x509|webhook
  authorization: {}
  bastion_host:
    ssh_agent_auth: false
  cloud_provider: {}
  ignore_docker_version: true
# 
# # Currently only nginx ingress provider is supported.
# # To disable ingress controller, set `provider: none`
# # To enable ingress on specific nodes, use the node_selector, eg:
#    provider: nginx
#    node_selector:
#      app: ingress
# 
  ingress:
    provider: nginx
  kubernetes_version: v1.15.0-rancher1-2
  monitoring:
    provider: metrics-server
# 
#   If you are using calico on AWS
# 
#    network:
#      plugin: calico
#      calico_network_provider:
#        cloud_provider: aws
# 
# # To specify flannel interface
# 
#    network:
#      plugin: flannel
#      flannel_network_provider:
#      iface: eth1
# 
# # To specify flannel interface for canal plugin
# 
#    network:
#      plugin: canal
#      canal_network_provider:
#        iface: eth1
# 
  network:
    plugin: weave
  restore:
    restore: false
# 
#    services:
#      kube-api:
#        service_cluster_ip_range: 10.43.0.0/16
#      kube-controller:
#        cluster_cidr: 10.42.0.0/16
#        service_cluster_ip_range: 10.43.0.0/16
#      kubelet:
#        cluster_domain: cluster.local
#        cluster_dns_server: 10.43.0.10
# 
  services:
    etcd:
      backup_config:
        enabled: true
        interval_hours: 12
        retention: 6
      creation: 12h
      extra_args:
        election-timeout: '5000'
        heartbeat-interval: '500'
      gid: 0
      retention: 72h
      snapshot: false
      uid: 0
    kube-api:
      always_pull_images: false
      pod_security_policy: false
      service_node_port_range: 30000-32767
    kube-controller: {}
    kubelet:
      fail_swap_on: false
    kubeproxy: {}
    scheduler: {}
  ssh_agent_auth: false

```

**Steps to Reproduce:**
```
docker run -it rancher/rke-tools:v0.1.40-arm64 bash -c "sha256sum /tmp/
loopback"
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  /tmp/loopback
```
```
docker run -it rancher/rke-tools:v0.1.40-amd64 bash -c "sha256sum /tmp/
loopback"
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  /tmp/loopback
```
```
curl -L https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-arm64-v0.4.0.tgz --output - | tar -xOzf - ./loopback | sha256sum
b58091e9e02ff50f75e0acb0e6c7591e9ba14c7983ef3f999f221e5471d2e85b  -
```
```
curl -L https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-amd64-v0.4.0.tgz --output - | tar -xOzf - ./loopback | sha256sum
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  -

```

**Results:** 
Incorrect binaries pulled into ARM64 image.

**Other details that may be helpful:**
This is keeping Weave from working correctly in my mixed architecture environment. When I manually move the correct architecture file into place, Weave is bootstrapped properly.

**Comment by westonmyers** [comment]
Same is seen in the latest `rancher/rke-tools`:

```
docker run -it rancher/rke-tools:v0.1.43-linux-arm64 bash -c "sha256sum /tmp/loopback"
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  /tmp/loopback
```
```
docker run -it rancher/rke-tools:v0.1.43-linux-amd64 bash -c "sha256sum /tmp/loopback"
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  /tmp/loopback
```

**Comment by westonmyers** [comment]
Built `rancher/rke-tools` on the current [master](https://github.com/rancher/rke-tools/tree/fca0192d733e86b1b433ca6994e9e60aa10c8253) via dapper.
```
docker run -it rancher/rke-tools:v0.1.43_arm64 bash -c "sha256sum /tmp/loopback"
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  /tmp/loopback
```
```
docker run -it rancher/rke-tools:v0.1.43_arm64 bash -c "sha256sum /tmp/loopback"
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  /tmp/loopback
```

So, two things. One, is that non-drone ci builds are forced to AMD64 via https://github.com/rancher/rke-tools/blob/fca0192d733e86b1b433ca6994e9e60aa10c8253/package/Dockerfile#L4

With that caveat known, the done ci builds should be fixed by changing 

https://github.com/rancher/rke-tools/blob/fca0192d733e86b1b433ca6994e9e60aa10c8253/package/Dockerfile#L20

from

`RUN wget -q -O - https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-amd64-v0.4.0.tgz | tar xzf - -C /tmp`

to

`RUN wget -q -O - https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-${ARCH}-v0.4.0.tgz | tar xzf - -C /tmp`

This is due to https://github.com/rancher/rke-tools/blob/fca0192d733e86b1b433ca6994e9e60aa10c8253/.drone.yml#L89

Of course, maybe this dockerfile needs to be "dapperized" so that this is a non-issue for non-drone ci builds.

**Comment by westonmyers** [comment]
Corrected in `rancher/rke-tools:v0.1.44`

```
curl -L https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-amd64-v0.4.0.tgz --output - | tar -xOzf - ./loopback | sha256sum
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  -
```
```
docker run -it rancher/rke-tools:v0.1.44-linux-amd64 bash -c "sha256sum /tmp/loopback"
7d4594bc2ad416f1b0133ac9ad66c5a23cd8c1f5f252b5c50102786674d5f41c  /tmp/loopback
```
```
curl -L https://github.com/containernetworking/cni/releases/download/v0.4.0/cni-arm64-v0.4.0.tgz --output - | tar -xOzf - ./loopback | sha256sum
b58091e9e02ff50f75e0acb0e6c7591e9ba14c7983ef3f999f221e5471d2e85b  -
```
```
docker run -it rancher/rke-tools:v0.1.44-linux-arm64 bash -c "sha256sum /tmp/loopb
ack"
b58091e9e02ff50f75e0acb0e6c7591e9ba14c7983ef3f999f221e5471d2e85b  /tmp/loopback
```

Obviously need RKE to utilize this new release, but this is already tagged for `v0.3.0`.

**Comment by westonmyers** [comment]
My understanding of the generation of releases was lacking.

I believe that as of commits https://github.com/rancher/kontainer-driver-metadata/commit/fe93aa460d63c2ef28cadd86321238b6217dc031 and https://github.com/rancher/kontainer-driver-metadata/commit/b813d099a1cdeb4c07fb0dc43b0c1d5ee871429a , that `rancher/rke-tools:v0.1.44` should be available "in the wild" soon.

The following `rancher/hyperkube` releases should be the cutover point.

`rancher/hyperkube:v1.13.10-rancher1-3`
`rancher/hyperkube:v1.14.6-rancher2-1`
`rancher/hyperkube:v1.15.3-rancher2-1`
`rancher/hyperkube:v1.16.0-beta.1-rancher1-1`

As of now, the images have not been pushed out to [Docker Hub](https://hub.docker.com/r/rancher/hyperkube/tags), so it's likely to be seen in a future alpha/beta release of `rancher/rancher:v2.3.0-x`, as commented in the first referenced commit.

So, with this being imminent, I'm closing.

Thanks for the work guys!
